R/utils-read.R

```diff
@@ -626,15 +626,15 @@ readEstimatedTimeSeries <- function(reportObject, seriesName, timezone, descript
         p <- estimatedPeriods[i,]
         startTime <- p$start
         endTime <- p$end
-        if(inverted){
-          estimatedSubset <- rbind(estimatedSubset, seriesData[['points']][seriesData[['points']][['time']] < startTime | seriesData[['points']][['time']] >= endTime,])
-        } else {
-          estimatedSubset <- rbind(estimatedSubset, seriesData[['points']][seriesData[['points']][['time']] >= startTime & seriesData[['points']][['time']] < endTime,])
-        }
+        estimatedSubset <- rbind(estimatedSubset, seriesData[['points']][seriesData[['points']][['time']] >= startTime & seriesData[['points']][['time']] < endTime,])
       }
     }
     #Replace data with only saved data
-    seriesData[['points']] <- estimatedSubset
+    if(inverted){
+      seriesData[['points']] <- seriesData[['points']][ !(seriesData[['points']][['time']] %in% estimatedSubset[['time']]), ]
```

Comment:
Actually, I kind of follow it. You're just pulling out the inverse scenario from the loop. So,

```r
for(i in 1:nrow(estimatedPeriods)) {
    p <- estimatedPeriods[i,]
    startTime <- p$start
    endTime <- p$end
    estimatedSubset_i <- subset(estimatedSubset, time >= startTime & time < endTime)
    estimatedSubset <- rbind(estimatedSubset, estimatedSubset_i)
}

if(inverted){
  nonEstimatedSubset <- subset(series[['points']], !time %in% estimatedSubset[['time']])
  series[['points']] <- nonEstimatedSubset
} else {
  series[['points']] <- estimatedSubset
}
```